File: Source/client-model/src/main/java/model/intention/DefendIntention.java
package model.intention;

import root.Point2D;
import root.model.event.ClientIntention;

public class DefendIntention extends ClientIntention {

	public String defenseType;

	public Point2D sourceField;
	public Point2D destinationField;

	public DefendIntention(String defenseType, String playerName, Point2D srcField, Point2D destField) {
		super(ClientIntentionType.Defend, playerName);

		this.defenseType = defenseType;
		this.sourceField = srcField;
		this.destinationField = destField;
	}

}

File: Source/server/rabbit-game/src/SharedModel/ModelEvents/DefendIntention.cs

namespace RabbitGameServer.SharedModel.ModelEvents
{
	public class DefendIntention : ClientIntention
	{

		public AttackType defenseType;

		public Point2D sourceField;
		public Point2D destinationField;

		public DefendIntention(string name,
			AttackType defenseType,
			Point2D sourceField,
			Point2D destinationField)
			: base(ClientIntentionType.Defend, name)
		{

			this.defenseType = defenseType;

			this.sourceField = sourceField;
			this.destinationField = destinationField;
		}
	}
}